README.md

```diff
@@ -1 +1,179 @@
-# graphql-lambda-terraform
\ No newline at end of file
+# graphql-lambda-terraform
+
+This project expose a graphql api for CRUD operations on Users.
+
+The graphql api is implemented using nodejs and graphql-yoga module. The persistance layer is implemented on AWS DynamoDB an connected to the nodejs app using the modele *@aws/dynamodb-data-mapper*.
+
+For deployment testing, the graphql api was deployed as lambda function on AWS and exposed through AWS API-Gateway
+
+# The overall architecture
+![](images/General-arch.png)
+
+# User model
+
+The api offer operations over an user. The properties this user has are:
+- id
+- name
+- dob (date of birth)
+- address
+- description
+- imageUrl
+- createAt
+- updateAt
+
+# Graphql operations
+## Queries
+- **findUser:** Find a user by Id
+- **findUsersByName:** Find users by name. This result set is paginated (by default by two items). Use *pageSize*(number) to set the number of items per page. Use *lastEvaluatedUser (userId, name)* to indicate from where the query should return items.
+- **getUsers:** Get all user. This result set is paginated (by default by two items). Use *pageSize*(number) to set the number of items per page. Use *lastEvaluatedId (userId)* to indicate from where the query should return items.
+- **getAddressCoordinates:** Gets the coordinates of the user address (use mapbox https://docs.mapbox.com/api/overview/)
+## Mutations
+- **createUser:** persist a new user
+- **updateUser:** update an existing user
+- **deleteUser:** delete an existing user
+
+# Graphql schema
+
+## Queries
+
+### findUser
+![](images/findUser.png)
+### findUsersByName
+![](images/findUsersByName.png)
+### getUsers
+![](images/getUsers.png)
+### getAddressCoordinates
+![](images/getAddressCoordinates.png)
+
+## Mutations
+
+### createUser
+![](images/createUser.png)
+### updateUser
+![](images/updateUser.png)
+### deleteUser
+![](images/deleteUser.png)
+# Setup and deployment
+
+## Enviroment variables
+
+In order to connect to AWS dynamoDB, the project require the following enviroment variables which are setup in the terraform lambda file (*[PROJECT_DIRECTORY]/terraform/lambda_graphql.tf*)
+- **ACCESS_KEY_ID:** your AWS account access key id.
+- **SECRET_ACCESS_KEY:** your AWS account secret access key
+- **NODE_ENV:** indicate the execution enviroment (for local testing, set it to development, anything else will be consider production)
+
+For this project to run, you need to deployed into AWS cloud, for that you are going to need the following.
+- AWS account
+- AWS CLI
+- Terraform
+
+## Compress source code
+
+Compress the *node_modules* and content of *dist/* folder and named as Archive.zip. Use the *scripts/compressFiles.sh* to generate the zip file
```

Comment:
Well, It is not necessary in this case, because that folder structure is created by the typescript compilation process "npm run build". Those scripts were build for this project only. But maybe it would be less guessing for developers, if I explain in the documentation (readme) why this folder structure exists.